File: FinalProject/ExchangeBook/Data/Notification.cs
using ExchangeBook.models;

namespace ExchangeBook.Data
{
    public class Notification
    {
        public int Id { get; set; }

        public int? InterestedId { get; set; }
        public virtual User? InterestedUser { get; set; }


        public int? UserId { get; set; }
        public virtual User? User { get; set; }

        public int? BookId { get; set; }
        public virtual Book? Book { get; set; }

        public NotificationType? Type { get; set; }
        public bool? IsSeen { get; set; }
    }
}

File: angular/book-Exchange/src/app/shared/interfaces/notification.ts
import { Book } from "./book";
import { User } from "./user";

export interface Notification{
id:number;
user:User;
book:Book;
isSeen:boolean;
}
export interface Notifica{
    id:number;
    interestedUser:User
    user:User;
    book:Book;
    notificationType:string,
    isSeen:boolean;
    }

    export interface Notificat{
        id:number;
        interestedUser:User
        user:User;
        book:Book;
        type:string,
        isSeen:boolean;
        }